File: Swift/LeetCode/LeetCode1833.swift
// https://leetcode.com/problems/maximum-ice-cream-bars

class Solution {
    func maxIceCream(_ costs: [Int], _ coins: Int) -> Int {
        var count = [Int](repeating: 0, count: costs.max()! + 1)
        for cost in costs {
            count[cost] += 1
        }
        var answer = 0
        var currentCoins = coins
        for cost in 1..<count.count {
            guard count[cost] != 0 else { continue }
            let canBuy = min(count[cost], currentCoins / cost)
            answer += canBuy
            currentCoins -= canBuy * cost
            if currentCoins < cost { break }
        }
        return answer
    }
}

File: Python/LeetCode/LeetCode1833.py
# https://leetcode.com/problems/maximum-ice-cream-bars

class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        count = [0] * (max(costs) + 1)
        for cost in costs:
            count[cost] += 1
        answer = 0
        for cost in range(1, len(count)):
            if count[cost] == 0:
                continue
            can_buy = min(count[cost], coins // cost)
            answer += can_buy
            coins -= can_buy * cost
            if coins < cost:
                break
        return answer
